### ISSUE TITLE
Validate label description length

### ISSUE THREAD

**Issue by 36degrees** [Financial-Times/github-label-sync#42]
## What

It'd be neat if the script was able to validate label description lengths.

## Details

Currently, if you have a label with a description greater than 100 characters (I think), the GitHub API will reject it, causing the script to exit in the middle of syncing with:

```
GitHub Error:
PATCH /repos/alphagov/design-system-team-internal/labels/effort%3A%20days
422: Validation Failed
```

Because this validation is happening when making the API requests, it won't be picked up when running github-label-sync with the `--dry-run` flag.

I understand you may not wish to implement validation in the script, in which case it'd be great if the validation errors were outputted as well as the `422: Validation Failed`, so that the user understands what needs fixing.

## Additional information

Thank you for open-sourcing this – we're trying it out over at https://github.com/alphagov/design-system-team-labels 🙌

**Comment by 36degrees** [comment]
Should have said, if this is something you'd be interested in, and you can give me some pointers on how you think it might best be done, I'd be happy to try and contribute it.

**Comment by rowanmanning** [comment]
Hi @36degrees, we'd be happy to include this! I think it makes sense, I'd rather output useful errors rather than a cryptic `Validation Failed`.

If you're interested in doing this work, I think adding a validation step above [this line](https://github.com/Financial-Times/github-label-sync/blob/master/lib/github-label-sync.js#L37) would make sense. That way we verify that the labels are valid before ever trying to access GitHub, potentially saving us from rate-limiting.

---

As an aside, we noticed your design system labels! I like what you're doing 🙂 in case it's of interest, we recently we made a change to the way we manage labels: we now have [a GitHub action](https://github.com/Financial-Times/origami-labels) in each repository. The upside is that we don't need to manage a token with access to all of our repos, the downside is we had to add some YAML to 90+ repositories 😄
